### ISSUE TITLE
The compilation of ocamlfind failed because "ocaml version 4.12.0 is too old"

### ISSUE THREAD

**Issue by annazolkieve** [ocaml/opam#4604]
```
#=== ERROR while installing ocamlfind.1.8.0 ===================================#
# opam-version 1.2.2
# os           linux
# command      ./configure -bindir /home/issta16/.opam/system/bin -sitelib /home/issta16/.opam/system/lib -mandir /home/issta16/.opam/system/man -config /home/issta16/.opam/system/lib/findlib.conf -no-custom -no-topfind
# path         /home/issta16/.opam/system/build/ocamlfind.1.8.0
# compiler     system (4.12.0)
# exit-code    1
# env-file     /home/issta16/.opam/system/build/ocamlfind.1.8.0/ocamlfind-17769-e7966f.env
# stdout-file  /home/issta16/.opam/system/build/ocamlfind.1.8.0/ocamlfind-17769-e7966f.out
# stderr-file  /home/issta16/.opam/system/build/ocamlfind.1.8.0/ocamlfind-17769-e7966f.err
### stdout ###
# Welcome to findlib version 1.8.0
# Configuring core...
# Checking for #remove_directory...
# Testing threading model...
# The -vmthread switch is not supported: Your ocaml version is too old!
```
Trying opam / ocaml for the 1st time. Does not work. Disappointed... What shall I do now?

**Comment by annazolkieve** [comment]
```
ocamlc -vmthread
The -vmthread argument of ocamlc is no longer supported
since OCaml 4.09.0.  Please switch to system threads, which have the
same API. Lightweight threads with VM-level scheduling are 
```

```
=-=- Installing compiler 4.07.0 -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default.comp] https://github.com/ocaml/ocaml/archive/4.07.0.tar.gz downloaded
Now compiling OCaml. This may take a while, please bear with us...
Processing: [4.07.0: make world]
```
Why compiling from source? Just give download & install the binaries.

**Comment by kit-ty-kate** [comment]
it seems you're using opam 1.2.2, a 6 years old and deprecated version of opam.
Here is a quick and easy way to upgrade to the latest version:
```
sh <(curl -sL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh)
```
see http://opam.ocaml.org/doc/Install.html in case you need more details

**Comment by dra27** [comment]
Sorry you've had a slightly bumpy start with OCaml, @annazolkieve! As @kit-ty-kate says, you've ended up with a very old version of OPAM which doesn't receive updates any more. Can I ask how you ended up with OPAM 1.2.2 but OCaml 4.12.0? I'm not aware of a distribution which has such an old version of OPAM at the same time as the latest OCaml, so I'm wondering if you've hit some very out-of-date instructions which we should update.

To answer your two questions:
1. The error you see in ocamlfind 1.8.0 was actually fixed in ocamlfind 1.8.1 but this version was released after the OPAM 1.2.2 package repository was "sunset". OCaml is 25 years old, so we do occasionally have to drop things which don't make sense any more - the vmthreads library came from a time when not every Unix system had a posix threads implementation.
2. The first sentence of our README is "Opam is a source-based package manager for OCaml" which is why it builds OCaml from sources! Binary packages are available via [opam-bin](https://ocamlpro.github.io/opam-bin/) (not supported directly by the opam team, but worked on by people who are very familiar with it or have worked on it before).

Finally, do please head over to our [Discuss forum](https://discuss.ocaml.org) where I hope you can be guided to a slightly less frustrating time with OCaml 🙂

**Comment by annazolkieve** [comment]
@dra27 Sorry for the delay. Thanks for the detailed answer. It turns out that the distribution was Ubuntu 16.04. The solution for my issues was to install Ubuntu 20.04 and then redo all the steps. Thanks for the support.
